packages/react-card/Spec.md

```diff
@@ -0,0 +1,403 @@
+
+### Non-interactive cards
+
+![Visual interaction of non-interactive cards using the mouse](./assets/non-interactive-interaction-mouse.png)
+
+Cards that have multiple actionable child components within them cannot be clickable. We preserve the use of `EnterKey` for moving focus to actionable controls inside the card, therefore the card surface cannot be used to set the state of the card but is reliant on the checkbox or similar inputs to set the card state.
+
+Mouse interaction
+
+- OnMouseEnter / Hover cursor style stays at default, rest state stays persistent no visual feedback is applied to the card surface
+- Hover & onClick (Checkbox) Hover state & pressed state applies to checkbox component
+- MouseUp (Checkbox) Card and Checkbox are both in selected state, clicking again toggles components to unselected state
+
+### Context menu
+
+![Visual interaction of the context menu on cards using the mouse](./assets/context-interaction-mouse.png)
+
+A context menu can be triggered by right clicking the surface or through an action in the card.
+
+1.Rest
+
+2. RightClick [LongPress]
+   Right click anywhere on card surface invokes content menu
+3. Show
+   Context menu appears close to cursor location on card
+4. Hide/Dismiss
+   Clicking or tapping anywhere outside of menu to dismiss
+
+#### Disabled
+
+![Visual interaction of disabled cards using the mouse](./assets/disabled.png)
+
+1 Unselected disabled
+
+1. Label: Selectable Card
+2. Role: Group
+3. State: Disabled, unselected
+
+2 Selected disabled
+
+1. Label: Selectable Card
+2. Role: Group
+3. State: Disabled, selected
+
+### Keyboard
+
+#### Selectable cards
+
+![Visual interaction of selectable cards using the keyboard](./assets/selectable-interaction-keyboard.png)
+
+1. Rest Default state is ‘unselected’
+2. Focus Focus rectangle is shown on card
+3. Focus Use spacebar to toggle state (selected/unselected)
+
+### Non-interactive cards
+
+![Visual interaction of non-interactive cards using the keyboard](./assets/non-interactive-interaction-keyboard.png)
+
+Keyboard interaction
+
+1. Focus Initial focus is set on card, to access items within the card use EnterKey or Spacebar
+2. Focus-Inside Focus moves to first actionable item in the card. Use ArrowKeys or TabKey to move focus to child elements within the card. To exit card use EscKey, to move focus back to card [parent]
+3. Checkbox-Focus Depending on your tab order of items within the card, the number of keystrokes will vary. Once focus is set on the checkbox by default it will be set to unselected
+4. SpaceBar or EnterKey will toggle both Card and Checkbox to selected state. Keying again will toggle to unselected
+
+### Context menu
+
+![Visual interaction of the context menu on cards using the keyboard](./assets/context-interaction-keyboard.png)
+
+1. Rest
+2. OnFocus - Focus rect. is set on card. Use shiftKey and press F10 to invoke context menu
+
+   `aria-haspopup="menu"`
```

Comment:
aria-haspopup suggest that a menu would appear on (left) click so it is not a good fit for indicating a context menu. In Teams, the pattern is to add a dummy element with "has context menu" text and use it in aria-describedby. Not ideal so open to other suggestions, but so far this was the best we could find.

Reply:
CCing @smhigley for discussion about this in particular